### PR TITLE
feat(client): improves video quality labeling

### PR DESCRIPTION
**Pull request by WuTofu**
This PR improves video quality labeling in the quality menu by switching from using only video height to using both width and height for video track identification and display. 

This introduces a new function to generate user-friendly resolution labels based on aspect ratio and resolution, and adds unit tests to ensure correct behavior across a wide range of video formats and edge cases.

### CODE REVIEW

File: client/src/util/misc.ts [hunk 1]
```diff
@@ -5,3 +5,154 @@ export function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O):
 export function isOfficialSite(): boolean {
 	return window.location.hostname.endsWith("opentogethertube.com");
 }
+
+function gcd(a: number, b: number): number {
+	return b === 0 ? a : gcd(b, a % b);
+}
```
Comment (dyc3) [inline client/src/util/misc.ts:18]: Would prefer this to not be recursive

File: client/src/components/composables/media-player.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export interface MediaPlayerWithPlaybackRate extends MediaPlayer {
 }
 
 export interface MediaPlayerWithQuality extends MediaPlayer {
-	getVideoTracks(): number[];
+	getVideoTracks(): { width: number; height: number }[];
```
Comment (dyc3) [inline client/src/components/composables/media-player.ts]: We should go ahead and make `{ width: number; height: number }` it's own named type.
